### ISSUE TITLE
Improper handling of log message format bytestrings

### ISSUE THREAD

**Issue by AA-Turner** [pylint-dev/pylint#10813]
### Bug description

Consider the following snippet:

```python
import logging

logging.critical(b'\xc0\xc0')
```

At runtime, this will print `CRITICAL:root:b'\xc0\xc0'` to stderr, as [`logging` calls `str(msg)`](https://github.com/python/cpython/blob/3.14/Lib/logging/__init__.py#L398) before interpolating with format arguments.

When running `pylint` over the same snippet, we fail with an F0002 fatal error, downstream of a `UnicodeDecodeError`. This is as the `LoggingChecker._check_format_string()` method has [special-case handling](https://github.com/pylint-dev/pylint/blob/7dee1f853261fcd30952e9bb84964b134ca47030/pylint/checkers/logging.py#L332-L333) for bytes, running:

https://github.com/pylint-dev/pylint/blob/7dee1f853261fcd30952e9bb84964b134ca47030/pylint/checkers/logging.py#L332-L333

This handling is incorrect; `pylint` should either run `format_string = str(format_string)` or choose a sensible option for `format_string.decode(errors=...)`.

A

### Command used

```shell
$ uvx pylint pylint_logging_format_bug.py
```

### Pylint output

```text
************* Module pylint_logging_format_bug
pylint_logging_format_bug.py:1:0: C0114: Missing module docstring (missing-module-docstring)
Exception on node <Call l.3 at 0x10951e920> in file '.../pylint_logging_format_bug.py'
Traceback (most recent call last):
  File ".../site-packages/pylint/utils/ast_walker.py", line 87, in walk
    callback(astroid)
  File ".../site-packages/pylint/checkers/logging.py", line 220, in visit_call
    self._check_log_method(node, name)
  File ".../site-packages/pylint/checkers/logging.py", line 261, in _check_log_method
    self._check_format_string(node, format_pos)
  File ".../site-packages/pylint/checkers/logging.py", line 333, in _check_format_string
    format_string = format_string.decode()
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc0 in position 0: invalid start byte
pylint_logging_format_bug.py:1:0: F0002: pylint_logging_format_bug.py: Fatal error while checking 'pylint_logging_format_bug.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in '.../pylint/pylint-crash-2026-01-16-08-18-58.txt'. (astroid-error)

-----------------------------------
Your code has been rated at 0.00/10
```

### Expected behavior

No exception or fatal error.

### Pylint version

```text
pylint 4.0.4
astroid 4.0.3
Python 3.10.19 (main, Oct  9 2025, 15:25:03) [Clang 17.0.0 (clang-1700.3.19.1)]
```

### OS / Environment

N/A

### Additional dependencies

N/A
